app/src/main/java/mozilla/org/webmaker/WebmakerApplication.java

```diff
@@ -30,10 +30,10 @@ public void onCreate() {
         Router.sharedRouter().setContext(getApplicationContext());
         Router.sharedRouter().map("/main", MainActivity.class);
         Router.sharedRouter().map("/main/:tab", MainActivity.class);
-        Router.sharedRouter().map("/map/:projectId", Project.class);
-        Router.sharedRouter().map("/projects/:projectId", Page.class);
-        Router.sharedRouter().map("/elements/:projectId/:elementId/:type", Element.class);
-        Router.sharedRouter().map("/projects/:projectId/elements/:elementId/:attributeId", Tinker.class);
+        Router.sharedRouter().map("/projects/:project", Project.class);
+        Router.sharedRouter().map("/projects/:project/pages/:page", Page.class);
```

Comment:
Not sure I 100% understand, but these match the API routes and we need each of these params to make API requests. Do you have an example of how you would structure this differently?